c_api/include/taichi/cpp/taichi.hpp

```diff
@@ -1,4 +1,5 @@
 // C++ wrapper of Taichi C-API
+#include <cstring>
 #include <list>
 #include <vector>
```

Comment:
nit: let's also include <utility> for std::exchange in a followup PR  as discussed offline.